generate-stats

```diff
@@ -100,6 +102,34 @@ def main():
     # Sort tables
     all_tables = sorted(all_tables, key=lambda x: x["data"][0][1], reverse=True)
 
+    # Select unexpected messages
+    messages = cursor.execute("""\
+            SELECT retry_reason, TestRuns.url
+            FROM Failures
+            JOIN TestRuns ON Failures.run = TestRuns.id
+            WHERE TestRuns.time > ? AND
+                retry_reason IS NOT null AND
+                retry_reason IS NOT 'be robust against unstable tests' AND
+                retry_reason IS NOT 'due to failure of test harness or framework';""", (since,)).fetchall()
+
+    messages = [m for m in messages if m[0] is not None]
+    # Replace any numbers > 100 with 0 (usually time stamps or PIDs)
+    messages = [(re.sub(r'\d\d[\d\.]+', '0', m[0]), m[1]) for m in messages]
```

Comment:
Totally unimportant, but `r\d{3,}` says "at least 3 repetitions" more directly.

Reply:
is decimal point included in `\d`? I don't think it is.

Reply:
Right, my bad.